.github/workflows/main.yml

```diff
@@ -11,30 +11,44 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-11, macos-12]
+        os: [macos-11, macos-12, ubuntu-latest]
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
         with:
           fetch-depth: 64
       - name: Cleanup /usr/local
+        if: startsWith(matrix.os,'macos')
         run: |
           sudo mkdir /opt/local.old
           sudo mv /usr/local/* /opt/local.old
+      - name: Install Linux Dependencies
+        if: startsWith(matrix.os,'ubuntu')
+        run: >
+          sudo apt update &&
+          sudo apt install mtree-netbsd
+          gnustep libcurl4-gnutls-dev
```

Comment:
The gnustep package isn't needed. Also it might make sense to use libcurl4-openssl-dev, since openssl is already installed in the image and gnutls is not?

Reply:
Thanks @jmroot. I've made the suggested changes.